### ISSUE TITLE
Role management - members count can be incorrect

### ISSUE THREAD

**Issue by StewartBellamy** [cloudscribe/cloudscribe#1005]
The members count on each role is incorrect, it is only looking at the user role table.  It should join user roles to the user table to get the correct count.  When a user is deleted the code does not tidy up their user roles (or kvp items) so there can be redundant user role rows.

This came to light on a system where we are importing users and the client was looking at the role counts to check that the import had worked.

![image](https://github.com/user-attachments/assets/d4f04694-b3ec-4f72-b4f7-ec60eb92f311)

**Comment by CrispinF** [comment]
I think there are two issues here: a) improving the count to be more reliable as suggested, and b) improving the clean-up on deletion of users. But we need to check the latter with @StewartBellamy as I thought this worked correctly under normal conditions, but maybe not - how were these users being deleted?

**Comment by StewartBellamy** [comment]
@StewartBellamy - I think they were being deleted through the UI but I can't be 100% sure, easy enough to test though.

**Comment by JimKerslake** [comment]
Duplicate of this:
https://github.com/cloudscribe/cloudscribe/issues/870

**Comment by markidox** [comment]
@StewartBellamy Are you able to give me some steps to replicate your issue please? Running CloudScribe and adding/removing roles/users worked as expected, including them being removed from the MsSql database.

**Comment by StewartBellamy** [comment]
@markidox - If you delete a user that has roles do the rows in the cs_userrole table also get removed?

**Comment by markidox** [comment]
@StewartBellamy Yes, just tried this with a user, added three roles, then deleted the user and all rows in the cs_UserRole table were also removed.

**Comment by JimKerslake** [comment]
How old is your Db?  Are you sure you haven't just got orphaned rows in there dating back from the time before this cascade deletion bug got fixed...?   I don't think we ever made provision for retrospective clean-up of cascade deletes that never happened in the past...     https://github.com/cloudscribe/cloudscribe/issues/870
